{"version":3,"file":"FatalError.test.js","sourceRoot":"","sources":["../../../../../src/private/node/ui/components/FatalError.test.tsx"],"names":[],"mappings":"AAAA,OAAO,EAAC,UAAU,EAAC,MAAM,iBAAiB,CAAA;AAC1C,OAAO,EAAC,QAAQ,EAAC,MAAM,mCAAmC,CAAA;AAC1D,OAAO,EAAC,UAAU,EAAE,QAAQ,EAAE,aAAa,EAAC,MAAM,kCAAkC,CAAA;AACpF,OAAO,EAAC,MAAM,EAAC,MAAM,qBAAqB,CAAA;AAC1C,OAAO,EAAC,QAAQ,EAAE,MAAM,EAAE,IAAI,EAAC,MAAM,QAAQ,CAAA;AAC7C,OAAO,KAAK,MAAM,OAAO,CAAA;AAEzB,QAAQ,CAAC,YAAY,EAAE,KAAK,IAAI,EAAE;IAChC,IAAI,CAAC,wDAAwD,EAAE,KAAK,IAAI,EAAE;QACxE,MAAM,KAAK,GAAG,IAAI,UAAU,CAAC,MAAM,EAAE,UAAU,CAAC,CAAA;QAChD,MAAM,EAAC,SAAS,EAAC,GAAG,MAAM,CAAC,oBAAC,UAAU,IAAC,KAAK,EAAE,KAAK,GAAI,CAAC,CAAA;QAExD,MAAM,CAAC,QAAQ,CAAC,SAAS,EAAG,CAAC,CAAC,CAAC,qBAAqB,CAAC;;;;;;;;;KASpD,CAAC,CAAA;IACJ,CAAC,CAAC,CAAA;IAEF,IAAI,CAAC,4CAA4C,EAAE,KAAK,IAAI,EAAE;QAC5D,MAAM,KAAK,GAAG,IAAI,UAAU,CAAC;YAC3B,mDAAmD;YACnD;gBACE,IAAI,EAAE;oBACJ,KAAK,EAAE;wBACL,uDAAuD;wBACvD,oDAAoD;qBACrD;iBACF;aACF;SACF,CAAC,CAAA;QAEF,MAAM,EAAC,SAAS,EAAC,GAAG,MAAM,CAAC,oBAAC,UAAU,IAAC,KAAK,EAAE,KAAK,GAAI,CAAC,CAAA;QAExD,MAAM,CAAC,QAAQ,CAAC,SAAS,EAAG,CAAC,CAAC,CAAC,qBAAqB,CAAC;;;;;;;;;KASpD,CAAC,CAAA;IACJ,CAAC,CAAC,CAAA;IAEF,IAAI,CAAC,8CAA8C,EAAE,KAAK,IAAI,EAAE;QAC9D,MAAM,KAAK,GAAG,IAAI,QAAQ,CAAC,kBAAkB,CAAC,CAAA;QAC9C,KAAK,CAAC,KAAK,GAAG;;;;;;KAMb,CAAA;QAED,MAAM,EAAC,SAAS,EAAC,GAAG,MAAM,CAAC,oBAAC,UAAU,IAAC,KAAK,EAAE,KAAK,GAAI,CAAC,CAAA;QAExD,MAAM,CAAC,QAAQ,CAAC,SAAS,EAAG,CAAC,CAAC,CAAC,qBAAqB,CAAC;;;;;;;;;;;;;KAapD,CAAC,CAAA;IACJ,CAAC,CAAC,CAAA;IAEF,IAAI,CAAC,2DAA2D,EAAE,KAAK,IAAI,EAAE;QAC3E,MAAM,KAAK,GAAG,IAAI,QAAQ,CAAC,kBAAkB,CAAC,CAAA;QAC9C,KAAK,CAAC,KAAK,GAAG;;;;;;KAMb,CAAA;QAED,KAAK,CAAC,SAAS,GAAG;YAChB;gBACE,UAAU;gBACV;oBACE,IAAI,EAAE;wBACJ,KAAK,EAAE,yCAAyC;wBAChD,GAAG,EAAE,qCAAqC;qBAC3C;iBACF;gBACD;oBACE,IAAI,EAAE,GAAG;iBACV;aACF;YACD,gEAAgE;YAChE;gBACE,gFAAgF;gBAChF;oBACE,OAAO,EAAE,SAAS;iBACnB;aACF;SACF,CAAA;QAED,MAAM,EAAC,SAAS,EAAC,GAAG,MAAM,CAAC,oBAAC,UAAU,IAAC,KAAK,EAAE,KAAK,GAAI,CAAC,CAAA;QAExD,MAAM,CAAC,QAAQ,CAAC,SAAS,EAAG,CAAC,CAAC,CAAC,qBAAqB,CAAC;;;;;;;;;;;;;;;;;;;;KAoBpD,CAAC,CAAA;IACJ,CAAC,CAAC,CAAA;IAEF,IAAI,CAAC,4EAA4E,EAAE,KAAK,IAAI,EAAE;QAC5F,MAAM,KAAK,GAAG,IAAI,QAAQ,CAAC,kBAAkB,CAAC,CAAA;QAC9C,KAAK,CAAC,KAAK,GAAG;;;;;;KAMb,CAAA;QAED,KAAK,CAAC,SAAS,GAAG;YAChB;gBACE,UAAU;gBACV;oBACE,IAAI,EAAE;wBACJ,KAAK,EAAE,yCAAyC;wBAChD,GAAG,EAAE,qCAAqC;qBAC3C;iBACF;gBACD;oBACE,IAAI,EAAE,GAAG;iBACV;aACF;YACD,gEAAgE;YAChE;gBACE,gFAAgF;gBAChF;oBACE,OAAO,EAAE,SAAS;iBACnB;aACF;SACF,CAAA;QAED,KAAK,CAAC,cAAc,GAAG;YACrB;gBACE,KAAK,EAAE,6BAA6B;gBACpC,IAAI,EAAE;oBACJ;wBACE,IAAI,EAAE;4BACJ,KAAK,EAAE,CAAC,kBAAkB,CAAC;yBAC5B;qBACF;oBACD;wBACE,IAAI,EAAE;4BACJ,KAAK,EAAE,mBAAmB;4BAC1B,KAAK,EAAE,CAAC,0BAA0B,CAAC;yBACpC;qBACF;iBACF;aACF;YACD;gBACE,KAAK,EAAE,+BAA+B;gBACtC,IAAI,EAAE;oBACJ;wBACE,IAAI,EAAE;4BACJ,KAAK,EAAE,CAAC,yBAAyB,CAAC;yBACnC;qBACF;iBACF;aACF;SACF,CAAA;QAED,MAAM,EAAC,SAAS,EAAC,GAAG,MAAM,CAAC,oBAAC,UAAU,IAAC,KAAK,EAAE,KAAK,GAAI,CAAC,CAAA;QAExD,MAAM,CAAC,QAAQ,CAAC,SAAS,EAAG,CAAC,CAAC,CAAC,qBAAqB,CAAC;;;;;;;;;;;;;;;;;;;;;;;;;;;;KA4BpD,CAAC,CAAA;IACJ,CAAC,CAAC,CAAA;IAEF,IAAI,CAAC,qCAAqC,EAAE,KAAK,IAAI,EAAE;QACrD,MAAM,KAAK,GAAG,IAAI,aAAa,CAAC,kBAAkB,EAAE,MAAM,EAAE,CAAC,SAAS,CAAC,CAAC,CAAA;QAExE,MAAM,EAAC,SAAS,EAAC,GAAG,MAAM,CAAC,oBAAC,UAAU,IAAC,KAAK,EAAE,KAAK,GAAI,CAAC,CAAA;QAExD,MAAM,CAAC,QAAQ,CAAC,SAAS,EAAG,CAAC,CAAC,CAAC,qBAAqB,CAAC;;;;;;;;;KASpD,CAAC,CAAA;IACJ,CAAC,CAAC,CAAA;AACJ,CAAC,CAAC,CAAA","sourcesContent":["import {FatalError} from './FatalError.js'\nimport {unstyled} from '../../../../public/node/output.js'\nimport {AbortError, BugError, ExternalError} from '../../../../public/node/error.js'\nimport {render} from '../../testing/ui.js'\nimport {describe, expect, test} from 'vitest'\nimport React from 'react'\n\ndescribe('FatalError', async () => {\n  test('renders correctly with a just a message and tryMessage', async () => {\n    const error = new AbortError('test', 'try this')\n    const {lastFrame} = render(<FatalError error={error} />)\n\n    expect(unstyled(lastFrame()!)).toMatchInlineSnapshot(`\n      \"╭─ error ──────────────────────────────────────────────────────────────────────╮\n      │                                                                              │\n      │  test                                                                        │\n      │                                                                              │\n      │  try this                                                                    │\n      │                                                                              │\n      ╰──────────────────────────────────────────────────────────────────────────────╯\n      \"\n    `)\n  })\n\n  test('renders correctly with a formatted message', async () => {\n    const error = new AbortError([\n      'There has been an error creating your deployment:',\n      {\n        list: {\n          items: [\n            'amortizable-marketplace-ext: Missing expected key(s).',\n            \"sub-ui-ext: You don't have access to this feature.\",\n          ],\n        },\n      },\n    ])\n\n    const {lastFrame} = render(<FatalError error={error} />)\n\n    expect(unstyled(lastFrame()!)).toMatchInlineSnapshot(`\n      \"╭─ error ──────────────────────────────────────────────────────────────────────╮\n      │                                                                              │\n      │  There has been an error creating your deployment:                           │\n      │    • amortizable-marketplace-ext: Missing expected key(s).                   │\n      │    • sub-ui-ext: You don't have access to this feature.                      │\n      │                                                                              │\n      ╰──────────────────────────────────────────────────────────────────────────────╯\n      \"\n    `)\n  })\n\n  test('renders correctly with a message and a stack', async () => {\n    const error = new BugError('Unexpected error')\n    error.stack = `\n      Error: Unexpected error\n          at Module._compile (internal/modules/cjs/loader.js:1137:30)\n          at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)\n          at Module.load (internal/modules/cjs/loader.js:985:32)\n          at Function.Module._load (internal/modules/cjs/loader.js:878:14)\n    `\n\n    const {lastFrame} = render(<FatalError error={error} />)\n\n    expect(unstyled(lastFrame()!)).toMatchInlineSnapshot(`\n      \"╭─ error ──────────────────────────────────────────────────────────────────────╮\n      │                                                                              │\n      │  Unexpected error                                                            │\n      │                                                                              │\n      │  To investigate the issue, examine this stack trace:                         │\n      │    at _compile (internal/modules/cjs/loader.js:1137)                         │\n      │    at js (internal/modules/cjs/loader.js:1157)                               │\n      │    at load (internal/modules/cjs/loader.js:985)                              │\n      │    at _load (internal/modules/cjs/loader.js:878)                             │\n      │                                                                              │\n      ╰──────────────────────────────────────────────────────────────────────────────╯\n      \"\n    `)\n  })\n\n  test('renders correctly with a message, a stack, and next steps', async () => {\n    const error = new BugError('Unexpected error')\n    error.stack = `\n      Error: Unexpected error\n          at Module._compile (internal/modules/cjs/loader.js:1137:30)\n          at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)\n          at Module.load (internal/modules/cjs/loader.js:985:32)\n          at Function.Module._load (internal/modules/cjs/loader.js:878:14)\n    `\n\n    error.nextSteps = [\n      [\n        'Have you',\n        {\n          link: {\n            label: 'created a Shopify Partners organization',\n            url: 'https://partners.shopify.com/signup',\n          },\n        },\n        {\n          char: '?',\n        },\n      ],\n      'Have you confirmed your accounts from the emails you received?',\n      [\n        'Need to connect to a different App or organization? Run the command again with',\n        {\n          command: '--reset',\n        },\n      ],\n    ]\n\n    const {lastFrame} = render(<FatalError error={error} />)\n\n    expect(unstyled(lastFrame()!)).toMatchInlineSnapshot(`\n      \"╭─ error ──────────────────────────────────────────────────────────────────────╮\n      │                                                                              │\n      │  Unexpected error                                                            │\n      │                                                                              │\n      │  Next steps                                                                  │\n      │    • Have you created a Shopify Partners organization [1]?                   │\n      │    • Have you confirmed your accounts from the emails you received?          │\n      │    • Need to connect to a different App or organization? Run the command     │\n      │      again with \\`--reset\\`                                                    │\n      │                                                                              │\n      │  To investigate the issue, examine this stack trace:                         │\n      │    at _compile (internal/modules/cjs/loader.js:1137)                         │\n      │    at js (internal/modules/cjs/loader.js:1157)                               │\n      │    at load (internal/modules/cjs/loader.js:985)                              │\n      │    at _load (internal/modules/cjs/loader.js:878)                             │\n      │                                                                              │\n      ╰──────────────────────────────────────────────────────────────────────────────╯\n      [1] https://partners.shopify.com/signup\n      \"\n    `)\n  })\n\n  test('renders correctly with a message, a stack, next steps, and custom sections', async () => {\n    const error = new BugError('Unexpected error')\n    error.stack = `\n      Error: Unexpected error\n          at Module._compile (internal/modules/cjs/loader.js:1137:30)\n          at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)\n          at Module.load (internal/modules/cjs/loader.js:985:32)\n          at Function.Module._load (internal/modules/cjs/loader.js:878:14)\n    `\n\n    error.nextSteps = [\n      [\n        'Have you',\n        {\n          link: {\n            label: 'created a Shopify Partners organization',\n            url: 'https://partners.shopify.com/signup',\n          },\n        },\n        {\n          char: '?',\n        },\n      ],\n      'Have you confirmed your accounts from the emails you received?',\n      [\n        'Need to connect to a different App or organization? Run the command again with',\n        {\n          command: '--reset',\n        },\n      ],\n    ]\n\n    error.customSections = [\n      {\n        title: 'amortizable-marketplace-ext',\n        body: [\n          {\n            list: {\n              items: ['Some other error'],\n            },\n          },\n          {\n            list: {\n              title: 'Validation errors',\n              items: ['Missing expected key(s).'],\n            },\n          },\n        ],\n      },\n      {\n        title: 'amortizable-marketplace-ext-2',\n        body: [\n          {\n            list: {\n              items: ['Something was not found'],\n            },\n          },\n        ],\n      },\n    ]\n\n    const {lastFrame} = render(<FatalError error={error} />)\n\n    expect(unstyled(lastFrame()!)).toMatchInlineSnapshot(`\n      \"╭─ error ──────────────────────────────────────────────────────────────────────╮\n      │                                                                              │\n      │  Unexpected error                                                            │\n      │                                                                              │\n      │  Next steps                                                                  │\n      │    • Have you created a Shopify Partners organization [1]?                   │\n      │    • Have you confirmed your accounts from the emails you received?          │\n      │    • Need to connect to a different App or organization? Run the command     │\n      │      again with \\`--reset\\`                                                    │\n      │                                                                              │\n      │  amortizable-marketplace-ext                                                 │\n      │    • Some other error                                                        │\n      │  Validation errors                                                           │\n      │    • Missing expected key(s).                                                │\n      │                                                                              │\n      │  amortizable-marketplace-ext-2                                               │\n      │    • Something was not found                                                 │\n      │                                                                              │\n      │  To investigate the issue, examine this stack trace:                         │\n      │    at _compile (internal/modules/cjs/loader.js:1137)                         │\n      │    at js (internal/modules/cjs/loader.js:1157)                               │\n      │    at load (internal/modules/cjs/loader.js:985)                              │\n      │    at _load (internal/modules/cjs/loader.js:878)                             │\n      │                                                                              │\n      ╰──────────────────────────────────────────────────────────────────────────────╯\n      [1] https://partners.shopify.com/signup\n      \"\n    `)\n  })\n\n  test('renders correctly an external error', async () => {\n    const error = new ExternalError('Unexpected error', 'yarn', ['install'])\n\n    const {lastFrame} = render(<FatalError error={error} />)\n\n    expect(unstyled(lastFrame()!)).toMatchInlineSnapshot(`\n      \"── external error ──────────────────────────────────────────────────────────────\n\n      Error coming from \\`yarn install\\`\n\n      Unexpected error\n\n      ────────────────────────────────────────────────────────────────────────────────\n      \"\n    `)\n  })\n})\n"]}